tests/experimental/datasources/test_integration_sql_datasource.py

```diff
@@ -163,3 +165,110 @@ def test_run_checkpoint_and_data_doc(empty_data_context, include_rendered_conten
     assert "ge-success-icon" in data_doc_index
     assert "ge-failed-icon" not in data_doc_index
+
+
+@pytest.mark.integration
+@pytest.mark.slow
+def test_run_data_assistant_and_checkpoint(empty_data_context):
+    """Test using data assistants to create expectation suite and run checkpoint"""
+    data_assistant_result, checkpoint_result = configure_and_run_data_assistant_checkpoint(
+        context=empty_data_context,
+        sqlite_db_file_relative_path="../../test_sets/taxi_yellow_tripdata_samples/sqlite/yellow_tripdata.db",
+        table_name="yellow_tripdata_sample_2019_01",
+        date_splitter_column="pickup_datetime",
+        batch_request_options={"year": 2019, "month": 1},
+    )
+    # We assert the data assistant successfully generated expectations.
+    # We don't care about the exact number since that may change as data assistants evolve.
+    expectation_num = len(data_assistant_result.expectation_configurations)
+    assert expectation_num > 100
+    assert checkpoint_result.success, "Running expectation suite failed"
+    # Verify that the number of checkpoint validations is the number of expectations generated by the data assistant
+    assert len(checkpoint_result["run_results"][list(checkpoint_result["run_results"].keys())[0]]["validation_result"].results) == expectation_num
+
+
+pytest.mark.integration
+@pytest.mark.slow
+def test_run_multibatch_data_assistant_and_checkpoint(empty_data_context):
+    """Test using data assistants to create expectation suite and run checkpoint"""
+    data_assistant_result, checkpoint_result = configure_and_run_data_assistant_checkpoint(
+        context=empty_data_context,
+        sqlite_db_file_relative_path="../../test_sets/taxi_yellow_tripdata_samples/sqlite/yellow_tripdata_sample_2020_all_months_combined.db",
+        table_name="yellow_tripdata_sample_2020",
+        date_splitter_column="pickup_datetime",
+        batch_request_options={"year": 2020},
+    )
+    expectation_num = len(data_assistant_result.expectation_configurations)
+    assert expectation_num > 100
+    assert checkpoint_result.success, "Running expectation suite failed"
+    # Verify that the number of checkpoint validations is the number of expectations generated by the data assistant
+    assert len(checkpoint_result["run_results"][list(checkpoint_result["run_results"].keys())[0]]["validation_result"].results) == expectation_num
+
+
+def configure_and_run_data_assistant_checkpoint(
+        context: DataContext,
+        sqlite_db_file_relative_path: str,
+        table_name: str,
+        date_splitter_column: str,
+        batch_request_options: Dict[str, Any]
+):
+    db_file = file_relative_path(__file__, sqlite_db_file_relative_path)
+    # Add sqlalchemy datasource.
+    # The current method is called `add_postgres` it really should be `add_sql`.
+    # See https://superconductive.atlassian.net/browse/GREAT-1400
```

Comment:
I thought the plan was to separate all SQL Alchemy dialects into separate datasource classes (thus decoupling these methods from the 3 `ExecutionEngine`s). The ticket says nothing is postgres-specific, but there is something that is sqlite-specific (where I commented above). If we elevate this to `add_sql` we probably should subclass the sqlite logic. If we intend to demo postgres I would think we would want to alias `add_postgres` to `add_sql` until something postgres-specific emerges. Validation of the connection string would be one example off the top of my head of something that will be postgres-specific.

Reply:
Yes, there will be a generic `add_sql` method for generic sql engines and then specific engines for postgres and sqlite. I will do this after this PR is in.

I only added the sqlite branch in this pr because I was using sqlite to test. It maybe would have been better to setup postgres to test but this was easier and worked in previous PRs.